main.go

```
@@ -1,6 +1,77 @@
 package main
-import "fmt"
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
 
 func main() {
-    fmt.Println("hello world")
+	type todo struct {
+		// set as true when edit or add button is pressed
+		EnableEdit bool   `false`
+		Id         string `json:"id"`
+		Title      string `json:"title"`
+		Completed  bool   `json:"completed"`
+		Action     string `ADD`
+	}
+
+	http.HandleFunc("/", handle)
+	http.HandleFunc("/", renderList)
+	http.HandleFunc("/", addTodo)
+	http.HandleFunc("/", deleteTodo)
+	http.HandleFunc("/", editTodo)
+}
+
+// takes input event, then renders the list associated to event
+// pass it to javascript to print out line by line
+func renderList() {
+	response, err := http.Get("https://api.coinbase.com/v2/prices/BTC-USD/buy")
+	if err != nil {
+		fmt.Printf("HTTP request failed with error %s\n", err)
+	} else {
+		data, _ := ioutil.ReadAll(response.Body)
+		fmt.Println(string(data))
+	}
+}
+
+// press button: edit, delete, add, complete
+// get request is just rendered automatically
+func handle(response http.ResponseWriter, request *http.Request) {
+	// s := todo{EnableEdit: false, Id: "123", Title: "Cultus Lake", Completed: false, Action: "ADD"}
+	switch request.Method {
+	case "ADD":
+		if this.EnableEdit == true {
+			response, _ = http.Put("https://httpbin.org/put", todoIndex, this.todo)
```

don't ignore errors! very dangerous - if this returns an error, attempting to access `response` will silently kill your program.

```go
response, err = http.Put("https://httpbin.org/put", todoIndex, this.todo)
if err != nil {
   http.Error(response, "request failed", http.StatusInternalServerError)
   return
}
```